graders/python/python_autograder/pl_execute.py

```diff
@@ -5,9 +5,25 @@
 import numpy.random
 import random
 from os.path import join
+from os.path import splitext
 from types import ModuleType, FunctionType
 from copy import deepcopy
 
+import io
+from IPython import get_ipython
+from nbformat import read
+from IPython.core.interactiveshell import InteractiveShell
+
+def extract_cell_content(f, ipynb_key):
+    nb = read(f, 4)
+    shell = InteractiveShell.instance()
+    content = ""
+    for cell in nb.cells:
+        if cell["cell_type"] == "code":
+            code = shell.input_transformer_manager.transform_cell(cell.source)
+            if ipynb_key in code:
```

Comment:
This will pass if `#grade` is _anywhere_ in the whole cell, I think? Or is that what you're wanting?

Reply:
originally I had only checking the first line, but then decided not to worry about it? I was getting the source code in a different way, which gave me lines, and when I changed I guess I didn't think of that anymore. I am not attached to any implementation, so I am open to suggestions

Reply:
It does feel like it would be safer to only look in the first line of the cell?
```
lines = code.splitlines()
first_line = lines[0] if len(lines) > 0 else ''
if ipynb_key in first_line:
```

Reply:
Scrap that, I think it would be better to do:
```
if code.strip().startswith(ipynb_key):
```

The current notebook actually says this ("The autograder will search for cells that **_start with_** a keyword").